cli/internal/terraform/loader_test.go

```diff
@@ -186,21 +186,16 @@ func TestPrepareUpgradeWorkspace(t *testing.T) {
 
 			path := path.Join(tc.pathBase, strings.ToLower(tc.provider.String()))
 
-			createFiles(t, file, tc.oldWorkspaceFiles, tc.oldWorkingDir)
-			createFiles(t, file, tc.newWorkspaceFiles, tc.newWorkingDir)
+			createFiles(t, file, tc.workspaceFiles, tc.workingDir)
 
-			err := prepareUpgradeWorkspace(path, file, tc.oldWorkingDir, tc.newWorkingDir, tc.backupDir)
+			err := prepareUpgradeWorkspace(path, file, tc.workingDir, tc.backupDir)
 
 			if tc.wantErr {
 				require.Error(err)
 			} else {
 				require.NoError(err)
 			}
-			checkFiles(t, file, func(err error) { assert.NoError(err) }, tc.newWorkingDir, tc.expectedFiles)
-			checkFiles(t, file, func(err error) { assert.NoError(err) },
-				tc.backupDir,
-				tc.oldWorkspaceFiles,
-			)
+			checkFiles(t, file, func(err error) { assert.NoError(err) }, tc.workingDir, tc.expectedFiles)
```

Comment:
could we check that the embedded file content is written and that the backupdir contains the old content?